### PR TITLE
LPS-181708 [EVP] Create Payment Confirmation Workflow

### PR DESCRIPTION
**Pull request by eliasantosf**
Task: [LPS-181708](https://issues.liferay.com/browse/LPS-181708)

### CODE REVIEW

**Comment by eliasantosf** [comment]
ci:test:sf

**Comment by eliasantosf** [comment]
ci:test:sf

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/src/main/resources/site-initializer/workflow-definitions/evp-payment-confirmation/workflow-definition.xml [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+ObjectEntry currentyObjectEntry = ObjectEntryLocalServiceUtil.getObjectEntry(classPK);
+
+Map < String, Serializable > currentObjectValues = currentyObjectEntry.getValues();
+
+JSONObject jsonObject = new JSONObject(currentObjectValues);
+
+long requestId = jsonObject.getLong("r_requestId_c_evpRequestId");
+
+ObjectEntry requestObjectEntry = ObjectEntryLocalServiceUtil.getObjectEntry(requestId);
```
Comment (nNilton) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/src/main/resources/site-initializer/workflow-definitions/evp-payment-confirmation/workflow-definition.xml:50]: rename to evpRequestObjectEntry

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/src/main/resources/site-initializer/workflow-definitions/evp-payment-confirmation/workflow-definition.xml [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+import com.liferay.object.model.ObjectEntry;
+import com.liferay.object.service.ObjectEntryLocalServiceUtil;
+import org.json.JSONObject;
+import java.io.Serializable;
+import java.util.Map;
+
+long classPK = GetterUtil.getLong((String) workflowContext.get(
+WorkflowConstants.CONTEXT_ENTRY_CLASS_PK));
+
+ObjectEntry currentyObjectEntry = ObjectEntryLocalServiceUtil.getObjectEntry(classPK);
```
Comment (nNilton) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/src/main/resources/site-initializer/workflow-definitions/evp-payment-confirmation/workflow-definition.xml:42]: typo

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/src/main/resources/site-initializer/workflow-definitions/evp-payment-confirmation/workflow-definition.xml [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+Map < String, Serializable > currentObjectValues = currentyObjectEntry.getValues();
```
Comment (nNilton) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/src/main/resources/site-initializer/workflow-definitions/evp-payment-confirmation/workflow-definition.xml:44]: rename to currentObjectEntryValues

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/src/main/resources/site-initializer/workflow-definitions/evp-payment-confirmation/workflow-definition.xml [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+
+JSONObject jsonObject = new JSONObject(currentObjectValues);
+
+long requestId = jsonObject.getLong("r_requestId_c_evpRequestId");
```
Comment (nNilton) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/src/main/resources/site-initializer/workflow-definitions/evp-payment-confirmation/workflow-definition.xml:48]: why you need to convert to json? you can simply use as a map and get the id and cast to long

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/src/main/resources/site-initializer/workflow-definitions/evp-payment-confirmation/workflow-definition.xml [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+
+ObjectEntry requestObjectEntry = ObjectEntryLocalServiceUtil.getObjectEntry(requestId);
+
+Map < String, Serializable > requestEntryValue = requestObjectEntry.getValues();
```
Comment (nNilton) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/src/main/resources/site-initializer/workflow-definitions/evp-payment-confirmation/workflow-definition.xml:52]: rename to evpRequestObjectEntryValues

**Comment by eliasantosf** [comment]
ci:test:bundle

**Review by nNilton — CHANGES_REQUESTED** [review]

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/src/main/resources/site-initializer/workflow-definitions/evp-payment-confirmation/workflow-definition.xml (hunk 2 above)
Comment (eliasantosf) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/src/main/resources/site-initializer/workflow-definitions/evp-payment-confirmation/workflow-definition.xml:42]: Não entendi esse comentário.

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/src/main/resources/site-initializer/workflow-definitions/evp-payment-confirmation/workflow-definition.xml (hunk 5 above)
Comment (eliasantosf) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/src/main/resources/site-initializer/workflow-definitions/evp-payment-confirmation/workflow-definition.xml]: Done!

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/src/main/resources/site-initializer/workflow-definitions/evp-payment-confirmation/workflow-definition.xml (hunk 4 above)
Comment (eliasantosf) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/src/main/resources/site-initializer/workflow-definitions/evp-payment-confirmation/workflow-definition.xml]: Done!

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/src/main/resources/site-initializer/workflow-definitions/evp-payment-confirmation/workflow-definition.xml (hunk 3 above)
Comment (eliasantosf) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/src/main/resources/site-initializer/workflow-definitions/evp-payment-confirmation/workflow-definition.xml]: Done!

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/src/main/resources/site-initializer/workflow-definitions/evp-payment-confirmation/workflow-definition.xml (hunk 2 above)
Comment (eliasantosf) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/src/main/resources/site-initializer/workflow-definitions/evp-payment-confirmation/workflow-definition.xml]: Done!

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/src/main/resources/site-initializer/workflow-definitions/evp-payment-confirmation/workflow-definition.xml (hunk 1 above)
Comment (eliasantosf) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/src/main/resources/site-initializer/workflow-definitions/evp-payment-confirmation/workflow-definition.xml]: Done!

**Comment by eliasantosf** [comment]
ci:test:sf

**Comment by nNilton** [comment]
ci:forward

**Pull request closed without merging**